src/masoniteorm/models/Model.py

```diff
@@ -709,9 +709,19 @@ def update_or_create(cls, wheres, updates):
         total.update(updates)
         total.update(wheres)
         if not record:
-            return self.create(total, id_key=cls.get_primary_key())
-
-        return self.where(wheres).update(total)
+            # if we don't return fresh, we don't get the primary_key that has been used,
+            # and we can't call it from outside the function lest we get a QueryBuilder.
+            #
+            # Without this we are reduced to performing a DIY update_or_create, e.g.:
+            #     ebay_order = EbayOrder.where({'order_id': d['order_id']}).first()
+            #     if not ebay_order:
+            #         ebay_order = EbayOrder.create(d).fresh()
+            #     else:
+            #         ebay_order.save()
+            return self.create(total, id_key=cls.get_primary_key()).fresh()
+
+        rv = self.where(wheres).update(total)
+        return self.where(wheres).first()
```

Comment:
@sfinktah i'm actually reverting this change. the wheres could not be unique enough i think for this to work. its possible we could do 

```
return self.where(total).first()
```

but it would have to be tested